plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -490,4 +518,66 @@ protected Session joinPushdownEnabled(Session session)
                 .setCatalogSessionProperty(session.getCatalog().orElseThrow(), "join_pushdown_enabled", "true")
                 .build();
     }
+
+    @Test(timeOut = 60_000)
+    public void testCancellation()
+            throws Throwable
+    {
+        if (!hasBehavior(SUPPORTS_CANCELLATION)) {
+            throw new SkipException("Cancellation is not supported by given connector");
+        }
+
+        Duration minimalQueryDuration = new Duration(1, MINUTES);
+
+        // verify that long lasting query is correct
+        assertQuerySucceeds("EXPLAIN " + getLongLastingQuery(minimalQueryDuration));
+
+        Set<RemoteDatabaseEvent> remoteEvents = captureRemoteEventsDuring(() -> {
+            String longLastingQuery = getLongLastingQuery(minimalQueryDuration);
+            Future<?> future = threadPool.submit(() -> assertQueryFails(longLastingQuery, "Query killed. Message: Killed by test"));
+            QueryId queryId = getQueryId(longLastingQuery);
+            sleep(1_000); // to make sure remote data source is queried
+            assertUpdate(format("CALL system.runtime.kill_query(query_id => '%s', message => '%s')", queryId, "Killed by test"));
+            sleep(1_000); // to make sure query is cancelled
+            future.get();
+        });
+
+        assertThat(remoteEvents).isNotEmpty();
+        assertThat(remoteEvents.stream())
+                .map(RemoteDatabaseEvent::getStatus)
+                .contains(CANCELLED);
+    }
+
+    private QueryId getQueryId(String query)
+    {
+        AtomicReference<QueryId> queryId = new AtomicReference<>();
+        assertEventually(() -> {
+            MaterializedResult queriesResult = getQueryRunner().execute(format(
+                    "SELECT query_id FROM system.runtime.queries WHERE state = 'RUNNING' AND query = '%s' AND query NOT LIKE '%%system.runtime.queries%%'",
```

Comment:
`state = 'RUNNING'` is redundant and may be masking a concurrency problem

We should make sure the queries created by `getLongLastingQuery` are unique.
(either by providing some pre-seeded random text, or by adding "Unique" to method name)